File: react native movies/cheat.py
import requests
import json
url ='https://api.themoviedb.org/3/discover/movie?api_key=10c0ab432b40d8e915f57ab49691bf04&sort_by=popularity.desc&primary_release_year='

r = requests.get(url)
l = []
id = 0
with open("helpers/years.js","w") as f:
    f.write("export const movieyears =")
    for i in range(2020,1882,-1):
        k = url + str(i)
        y = requests.get(k).json()["results"][0]
        id += 1
        print(i)
        l.append({"poster":f"https://image.tmdb.org/t/p/w200{y['poster_path']}","key":id,"year":i})
    f.write(str(l))

File: react native movies/cheat.js
import * as React from 'react';

const apikey = "10c0ab432b40d8e915f57ab49691bf04"
let id = 0
const render = (movie) => {
return ({
  poster:`https://image.tmdb.org/t/p/w200/${movie.poster_path}`,
  year:movie.release_date.substring(0,4),
  key:id++,
}
)
}
async function year(){
    let jim = []
    for(let i = 1883; i <= 2020;i++){
    const movie = await fetch(`https://api.themoviedb.org/3/discover/movie?api_key=10c0ab432b40d8e915f57ab49691bf04"&primary_release_year${i}&sort_by=popularity.desc`)
    const list = await movie.json()
    jim.push(list.results.map(render)[0])
  }
  console.log(JSON.stringify(jim))
}


year()
